presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduClientSession.java

```diff
@@ -146,13 +147,14 @@ public Schema getTableSchema(KuduTableHandle tableHandle)
         return KuduTableProperties.toMap(table);
     }
 
-    public List<KuduSplit> buildKuduSplits(KuduTableHandle tableHandle)
+    public List<KuduSplit> buildKuduSplits(KuduTableHandle tableHandle, DynamicFilter dynamicFilter)
     {
         KuduTable table = tableHandle.getTable(this);
         int primaryKeyColumnCount = table.getSchema().getPrimaryKeyColumnCount();
         KuduScanToken.KuduScanTokenBuilder builder = client.newScanTokenBuilder(table);
 
-        TupleDomain<ColumnHandle> constraint = tableHandle.getConstraint();
+        TupleDomain<ColumnHandle> constraint = tableHandle.getConstraint()
+                .intersect(dynamicFilter.getCurrentPredicate());
```

Comment:
do we simplify domains somewhere?

Reply:
Modified

Reply:
I don't see any simplification of domains taking place in Kudu connector, so we don't have any point of reference for how big a predicate Kudu can handle.
I would suggest
```
TupleDomain<ColumnHandle> constraint = tableHandle.getConstraint()
                .intersect(dynamicFilter.getCurrentPredicate().simplify(100));
```
This way we don't change existing behaviour without DF and put a reasonable limit on DFs.

Reply:
Modified